lib/ricohapi/mstorage/client.rb

```diff
@@ -90,7 +92,18 @@ def delete(media_id)
 
       # PUT /media/{id}/meta/user/{key}
       def add_meta(media_id, user_meta)
-        # TODO: do something
+        if user_meta.empty?
+          raise Error.new("user_meta is empty: nothing to request.")
+        end
+        user_meta.each do |k, v|
+          if k.empty? || !valid?(v)
+            raise Error.new("Invalid parameter: {#{k} => #{v}}")
+          end
```

Comment:
I think you should raise the error in `valid?(v)` method.
ex)
```ruby
def valid?(value)
  raise Error.new("Invalid parameter: {#{k} => #{v}}") unless (MIN_USER_META_LENGTH..MAX_USER_META_LENGTH).include? value.length
  true
end 
``` 
And I think the method name should be `validate(value)`.